go/cmd/ct-fetch/ct-fetch.go

```diff
@@ -278,100 +448,240 @@ func (ld *LogSyncEngine) NewLogWorker(ctLogUrl string) (*LogWorker, error) {
 		return nil, err
 	}
 
-	var startPos uint64
-	// Now we're OK to use the DB
-	if *ctconfig.Offset > 0 {
-		glog.Infof("[%s] Starting from offset %d", ctLogUrl, *ctconfig.Offset)
-		startPos = *ctconfig.Offset
+	batchSize := *ctconfig.BatchSize
+
+	var task LogWorkerTask
+	if sth.TreeSize <= 3 {
+		// For technical reasons, we can't verify our download
+		// until there are at least 3 entries in the log. So
+		// we'll wait.
+		task = Sleep
+	} else if logObj.LastUpdateTime.IsZero() {
+		// First contact with log
+		task = Init
+	} else if logObj.MaxEntry < sth.TreeSize-batchSize {
+		// There are many new entries to download.
+		task = Update
+	} else if logObj.MinEntry > 0 {
```

Comment:
Is `MinEntry > 0` right here? Won't that be true almost always, after we've done some downloading?

Reply:
It downloads in reverse order, so MinEntry > 0 is true until we've read all the way back to the beginning of the log.

Reply:
Ohh, right - ok.